src/Microsoft.Extensions.Configuration/ConfigurationProvider.cs

```diff
@@ -32,21 +32,20 @@ public virtual void Load()
        
         public virtual IEnumerable<string> GetChildKeys(
             IEnumerable<string> earlierKeys,
-            string parentPath,
-            string delimiter)
+            string parentPath)
         {
-            var prefix = parentPath == null ? string.Empty : parentPath + delimiter;
+            var prefix = parentPath == null ? string.Empty : parentPath + ConfigurationPath.KeyDelimiter;
 
             return Data
                 .Where(kv => kv.Key.StartsWith(prefix, StringComparison.OrdinalIgnoreCase))
-                .Select(kv => Segment(kv.Key, prefix, delimiter))
+                .Select(kv => Segment(kv.Key, prefix))
                 .Concat(earlierKeys)
                 .OrderBy(k => k, ConfigurationKeyComparer.Instance);
         }
 
-        private static string Segment(string key, string prefix, string delimiter)
+        private static string Segment(string key, string prefix)
```

Comment:
While you are at it, could you change this little method to take `prefix.Lenght` instead of `prefix`? It is not super important, but it is a small improvement :smile: